### PR TITLE
Replace all formatted strings by f-strings

### LINKED ISSUE #21: Use f-strings for `format_output_str` methods
Minor refactoring: The code will become more readable if we use f-strings instead of the `" ".format()` syntax in the `format_output_str` methods of `CGPNode`.

### PR DESCRIPTION
**Pull request by mschmidt87**
As the PR title says: it simply replaces all formatted strings by f-strings, which is the more modern and readable syntax.

Fixes #21 .

### CODE REVIEW

File: gp/cgp_graph.py [hunk 1]
```diff
@@ -179,8 +179,8 @@ def __getitem__(self, key):
     def to_str(self):
 
         self._format_output_str_of_all_nodes()
-
-        return '[{}]'.format(', '.join(node.output_str for node in self.output_nodes))
+        out_str = ', '.join(node.output_str for node in self.output_nodes)
+        return f'[{out_str}]'
```
Comment (jakobj) [inline gp/cgp_graph.py]: would this work?
```suggestion
        return f'[{", ".join(node.output_str for node in self.output_nodes)}]'
```

File: gp/cgp_graph.py [hunk 2]
```diff
@@ -204,8 +204,8 @@ def to_func(self):
             Callable executing the function represented by the computational graph.
         """
         self._format_output_str_of_all_nodes()
-
-        func_str = 'def _f(x): return [{}]'.format(', '.join(node.output_str for node in self.output_nodes))
+        s = ', '.join(node.output_str for node in self.output_nodes)
+        func_str = 'def _f(x): return [{s}]'
```
Comment (jakobj) [inline gp/cgp_graph.py]: (could remove `s` like above)

File: gp/cgp_graph.py [hunk 3]
```diff
@@ -274,7 +272,7 @@ def update_parameters_from_torch_class(self, torch_cls):
         for n in self._nodes:
             if n.is_parameter:
                 try:
-                    n._output = eval('torch_cls._p{}[0]'.format(n._idx))
+                    n._output = eval('torch_cls._p{n._idx}[0]')
```
Comment (jakobj) [inline gp/cgp_graph.py]:
```suggestion
                    n._output = eval(f'torch_cls._p{n._idx}[0]')
```

File: gp/cgp_node.py [hunk 4]
```diff
@@ -189,7 +189,7 @@ def __call__(self, x, graph):
         self._output = graph[self._inputs[0]].output * graph[self._inputs[1]].output
 
     def format_output_str(self, graph):
-        self._output_str = '({} * {})'.format(graph[self._inputs[0]].output_str, graph[self._inputs[1]].output_str)
+        self._output_str = '({graph[self._inputs[0]].output_str} * {graph[self._inputs[1]].output_str})'
```
Comment (jakobj) [inline gp/cgp_node.py]:
```suggestion
        self._output_str = f'({graph[self._inputs[0]].output_str} * {graph[self._inputs[1]].output_str})'
```

File: gp/cgp_node.py [hunk 5]
```diff
@@ -224,13 +223,13 @@ def __call__(self, x, graph):
         pass
 
     def format_output_str(self, graph):
-        self._output_str = '{}'.format(self._output)
+        self._output_str = f'{self._output}'
 
     def format_output_str_torch(self, graph):
-        self._output_str = 'self._p{}.expand(x.shape[0])'.format(self._idx)
+        self._output_str = 'self._p{self._idx}.expand(x.shape[0])'
```
Comment (jakobj) [inline gp/cgp_node.py]:
```suggestion
        self._output_str = f'self._p{self._idx}.expand(x.shape[0])'
```

File: gp/cgp_graph.py (hunk 1 above)
Comment (mschmidt87) [reply gp/cgp_graph.py]: Yes, but I found it kind of ugly and unreadble, too many brackets and quotation marks. But if you prefer this, it's fine for me.

File: gp/cgp_graph.py (hunk 2 above)
Comment (mschmidt87) [reply gp/cgp_graph.py]: yeah, same as above.

File: gp/cgp_graph.py (hunk 1 above)
Comment (jakobj) [reply gp/cgp_graph.py]: ok, let's keep it your way. we're writing for humans after all :)

File: gp/cgp_graph.py (hunk 2 above)
Comment (jakobj) [reply gp/cgp_graph.py]: ok

File: gp/abstract_individual.py [hunk 6]
```diff
@@ -19,7 +19,7 @@ def __init__(self, fitness, genome):
         self.idx = None  # an identifier to keep track of all unique genomes
 
     def __repr__(self):
-        return 'Individual(idx={}, fitness={}, genome={}))'.format(self.idx, self.fitness, self.genome)
+        return 'Individual(idx={self.idx}, fitness={self.fitness}, genome={self.genome}))'
```
Comment (jakobj) [inline gp/abstract_individual.py]:
```suggestion
        return f'Individual(idx={self.idx}, fitness={self.fitness}, genome={self.genome}))'
```

File: gp/cgp_graph.py (hunk 2 above)
Comment (jakobj) [inline gp/cgp_graph.py]:
```suggestion
        func_str = f'def _f(x): return [{s}]'
```

File: gp/cgp_graph.py [hunk 7]
```diff
@@ -245,11 +245,9 @@ def __init__(self):
         func_str = \
 """
```
Comment (jakobj) [inline gp/cgp_graph.py]:
```suggestion
f"""
```

**Pull request merged**